File: 3354-make-array-elements-equal-to-zero/3354-make-array-elements-equal-to-zero.js
var countValidSelections = function (nums) {
    
    let sum = nums.reduce((a, b) => a + b, 0);

    let count = 0;
    let rem = 0;

    for (let i = 0; i < nums.length; i++) {
        if (nums[i] == 0) {
            let diff = sum - rem;

            if (rem == diff)
                count += 2;
            else if (Math.abs(rem - diff) == 1)
                count++
        }
        rem += nums[i];
    }
    return count;
};

File: 3354-make-array-elements-equal-to-zero/3354-make-array-elements-equal-to-zero.java
class Solution {
    public int countValidSelections(int[] nums) {
        int sum = 0;
        for (int n : nums) sum += n;

        int count = 0;
        int rem = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0) {
                int diff = sum - rem;
                if (diff == rem)
                    count += 2;
                else if (Math.abs(diff - rem) == 1)
                    count++;
            }
            rem += nums[i];
        }
        
        return count;
    }
}